tar/chunks/chunks_stats.c

```diff
@@ -335,8 +335,8 @@ chunks_stats_addchunk(CHUNKS_S * C, const uint8_t * hash,
 
 	/* Record the lengths if necessary. */
 	if (ch->d.nrefs == 0 && ch->ncopies_ctape == 0) {
-		ch->d.len = len;
-		ch->d.zlen_flags = zlen | (ch->d.zlen_flags & CHDATA_FLAGS);
+		ch->d.len = (uint32_t)len;
```

Comment:
Maybe add an assertion here?  In general assertions are good for "this is safe because all the users of this code behave" situations since they catch if a new use of the code misbehaves.  (Or, put another way, we should use assertions to check that APIs are being used correctly, and chunks_* is an API.)